src/modules/iteratearray.js

```diff
@@ -1,20 +1,80 @@
+/* eslint-disable no-use-before-define */
+
 const listelementcontainer = document.querySelector('.listelementcontainer');
 
+const switchelement = (inputelement, elementcontainer, arr, i) => {
+  const newlistelement = document.createElement('div');
+  newlistelement.classList.add('listelement');
+  newlistelement.innerText = inputelement.value;
+  inputelement.parentNode.replaceChild(newlistelement, inputelement);
+  newlistelement.addEventListener('dblclick', () => {
+    editelement(newlistelement, elementcontainer, arr, i);
+  });
+};
+
+const editelement = (listelement, elementcontainer, arr, i) => {
+  const threedots = elementcontainer.querySelector('.threedotsicon');
+  const deletebutton = elementcontainer.querySelector('.deleteicon');
+  deletebutton.classList.remove('dnone');
+  threedots.classList.add('dnone');
+  const newinput = document.createElement('input');
+  newinput.type = 'text';
+  newinput.classList.add('yellowbg');
+  newinput.value = listelement.innerText;
+  listelement.parentNode.replaceChild(newinput, listelement);
+  newinput.focus();
+  newinput.select();
+  let deletebuttonpressed = false;
+  deletebutton.addEventListener('click', () => {
+    deletebuttonpressed = true;
+    arr.splice(i, 1);
+    for (let j = i; j < arr.length; j += 1) {
+      arr[j].index = j + 1;
+    }
+    localStorage.setItem('ToDoList', JSON.stringify(arr));
+    iteratearray(arr);
```

Comment:
- [ ] The delete functionality is still not working. Kindly make the delete button remove the item as demanded by the project requirements.